cmake/onnxruntime_mlas.cmake

```diff
@@ -281,6 +281,9 @@ function(setup_kleidiai)
     ${MLAS_SRC_DIR}/kai_ukernel_interface.cpp
   )
   target_link_libraries(onnxruntime_mlas PRIVATE kleidiai)
+
+  list(APPEND onnxruntime_EXTERNAL_LIBRARIES kleidiai)
```

Comment:
Q. If this is important for Apple, should this be done conditionally?

Reply:
I'm not sure. `onnxruntime_EXTERNAL_LIBRARIES` is also used elsewhere (like some `target_link_libraries()` calls), but I can only say that this is required for the Apple static framework to work. it may be required in those other places too but I haven't confirmed that.